mirascope/core/openai/stream.py

```diff
@@ -39,13 +45,60 @@ def cost(self) -> float | None:
 
     def _construct_message_param(
         self,
-        tool_calls: list[ChatCompletionMessageToolCallParam] | None = None,
+        tool_calls: list[ChatCompletionMessageToolCall] | None = None,
         content: str | None = None,
     ) -> ChatCompletionAssistantMessageParam:
-        message_param = ChatCompletionAssistantMessageParam(
+        message_param = ChatCompletionMessage(
             role="assistant",
             content=content,
         )
         if tool_calls:
-            message_param["tool_calls"] = tool_calls
-        return message_param
+            message_param.tool_calls = tool_calls
+        return message_param.model_dump()  # type: ignore
+
+    def construct_call_response(self) -> OpenAICallResponse:
+        if self.message_param is None:
+            raise ValueError(
+                "No stream response, check if the stream has been consumed."
+            )
+        stream_tool_calls = self.message_param.get("tool_calls", [])
+        message = ChatCompletionMessage(
+            content=self.message_param.get("content", ""),
+            role=self.message_param["role"],
+        )
+        if stream_tool_calls:
+            tool_calls = []
+            for stream_tool_call in stream_tool_calls:
+                function_dict = stream_tool_call["function"]
+                tool_calls.append(
+                    ChatCompletionMessageToolCall(
+                        id=stream_tool_call["id"],
+                        function=Function(
+                            arguments=function_dict["arguments"],
+                            name=function_dict["name"],
+                        ),
+                        type="function",
+                    )
+                )
+            message.tool_calls = tool_calls
+        completion = ChatCompletion(
+            id="id",
+            model=self.model,
+            choices=[Choice(finish_reason="stop", index=0, message=message)],
```

Comment:
shouldn't you just be able to set this to `ChatCompletionMessage(**self.message_param)` and call it a day?

Reply:
oh oops this was leftover code, updated